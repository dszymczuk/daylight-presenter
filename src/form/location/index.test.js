import React from "react";
import { render, cleanup } from '@testing-library/react';
import { AddForm, EditForm } from "./index";
import LocationContextProvider from "../../contexts/locations";

afterEach(cleanup);

it("renders AddForm", () => {
  const { asFragment } = render(
    <LocationContextProvider>
      <AddForm/>
    </LocationContextProvider>);
  expect(asFragment()).toMatchSnapshot();
});

it("renders EditForm", () => {
  const props = {
    initialValues: {
      name: "Sample location",
      lat: 20,
      lng: 30,
      date: new Date()
    },
    index: 0,
    onSubmitCb: () => {
    }
  };
  const { asFragment } = render(
    <LocationContextProvider>
      <EditForm {...props} />
    </LocationContextProvider>
  );
  expect(asFragment()).toMatchSnapshot();
});