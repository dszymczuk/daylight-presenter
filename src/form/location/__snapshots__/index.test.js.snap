// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders AddForm 1`] = `
<DocumentFragment>
  <div>
    <form
      class=""
    >
      <div
        class="card"
      >
        <h5
          class="card-header"
        >
          <span>
            Add new location
          </span>
        </h5>
        <div
          class="card-body"
        >
          <div
            class="row"
          >
            <div
              class="col"
            >
              <div
                class="form-group"
              >
                <label
                  class="form-label"
                  for="name"
                >
                  Name
                </label>
                <input
                  class="form-control"
                  id="name"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="col"
            >
              <div
                class="form-group"
              >
                <label
                  class="form-label"
                  for="lat"
                >
                  Lat
                </label>
                <input
                  class="form-control"
                  id="lat"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="col"
            >
              <div
                class="form-group"
              >
                <label
                  class="form-label"
                  for="lng"
                >
                  Lng
                </label>
                <input
                  class="form-control"
                  id="lng"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="col"
            >
              <div
                class="datepickerWrapper"
              >
                <div
                  class="form-group"
                >
                  <label
                    class="form-label"
                    for="date"
                  >
                    Date
                  </label>
                  <div
                    class="react-datepicker-wrapper"
                  >
                    <div
                      class="react-datepicker__input-container"
                    >
                      <input
                        class="form-control"
                        id="date"
                        value="26.01.2020"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="row"
          >
            <div
              class="col"
            >
              <button
                class="btn btn-success btn-block"
                type="submit"
              >
                Add location
              </button>
            </div>
          </div>
        </div>
      </div>
    </form>
  </div>
</DocumentFragment>
`;

exports[`renders EditForm 1`] = `
<DocumentFragment>
  <div>
    <form
      class=""
    >
      <div
        class="card"
      >
        <h5
          class="card-header"
        >
          <span>
            Edit Sample location location
          </span>
        </h5>
        <div
          class="card-body"
        >
          <div
            class="row"
          >
            <div
              class="col"
            >
              <div
                class="form-group"
              >
                <label
                  class="form-label"
                  for="name"
                >
                  Name
                </label>
                <input
                  class="form-control"
                  id="name"
                  type="text"
                  value="Sample location"
                />
              </div>
            </div>
            <div
              class="col"
            >
              <div
                class="form-group"
              >
                <label
                  class="form-label"
                  for="lat"
                >
                  Lat
                </label>
                <input
                  class="form-control"
                  id="lat"
                  type="text"
                  value="20"
                />
              </div>
            </div>
            <div
              class="col"
            >
              <div
                class="form-group"
              >
                <label
                  class="form-label"
                  for="lng"
                >
                  Lng
                </label>
                <input
                  class="form-control"
                  id="lng"
                  type="text"
                  value="30"
                />
              </div>
            </div>
            <div
              class="col"
            >
              <div
                class="datepickerWrapper"
              >
                <div
                  class="form-group"
                >
                  <label
                    class="form-label"
                    for="date"
                  >
                    Date
                  </label>
                  <div
                    class="react-datepicker-wrapper"
                  >
                    <div
                      class="react-datepicker__input-container"
                    >
                      <input
                        class="form-control"
                        id="date"
                        value="26.01.2020"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="row"
          >
            <div
              class="col"
            >
              <button
                class="btn btn-warning btn-block"
                type="submit"
              >
                Edit location
              </button>
            </div>
          </div>
        </div>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
